server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileProjectOperations.java

```diff
@@ -77,8 +77,9 @@ private QualityProfileDto selectProfileByKey(DbSession session, String profileKe
     return WsUtils.checkFound(qualityProfile, "Quality profile does not exist");
   }
 
-  private void checkAdminOnProject(String projectKey) {
-    if (!userSession.hasPermission(GlobalPermissions.QUALITY_PROFILE_ADMIN) && !userSession.hasComponentPermission(UserRole.ADMIN, projectKey)) {
+  private void checkAdminOnProject(ComponentDto project) {
+    if (!userSession.hasPermission(GlobalPermissions.QUALITY_PROFILE_ADMIN) &&
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "hasPermission"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)